### ISSUE TITLE
Feature Request: support for mount/one-direction sync in clients

### ISSUE THREAD

**Issue by ingoj** [nextcloud/desktop#267]
As request in discussion in https://twitter.com/ingoj/status/988826864146944000 here's the Feature Request: 

Please add a way to have some sort of "mounting" or "one-direction sync" in Nextcloud client such as on macOS. 

I like the idea to have one single virtual drive of "Nextcloud" in my Finder/Explorer, but being able to access all of my files on my Nextcloud server. Selecting all directory on small sized PCs/clients might fill up the local PC disk drive. 
When being able to upload new stuff via the client to the Nextcloud server, without the need of a local copy as it is the case with synced folders, the clients could save lots of local storage space while having the flexibility to access (download on the fly as needed) from Nextcloud. 

Basically there is something similar in Nextclouds mobile clients like on iOS where you can sync your full-sized photos to the cloud and access these photos on the iPhone in a smaller resolution. When viewing the full sized image will then be downloaded to the phone and gets displayed. 

Imagine a laaaarge personal cloud storage in Nextcloud that you want to seamlessly access from a small sized netbook or similar. Multiple TB in data just doesn't fit on a 128 GB SSD. 

In the discussion on Twitter there was WebDAV mentioned. Maybe this could be done by using WebDAV, but then again it is not a seamless experience in a single Nextcloud virtual drive, which I find very comfortable to use. :-)

**Comment by k1l1** [comment]
The same like Files on demand #17 or this owncloud/client#4341 ?

**Comment by ingoj** [comment]
Yes, maybe... but I don't know Google File Stream, so cannot comment on this. ;)
But yes, for #4341 this sounds similar. It is more or less a virtual filesystem, with the option to sync some files or directory to a local copy to increase access times and speed, maybe. 
In #4341 OpenAFS is mentioned. I was thinking about whether or not mentioning GlusterFS support when I wrote #267.

In the end I believe all 3 issues are aiming for the same target: having the ability to access all files in NC cloud without the need to download/sync them all. #4341 is most likely the same what I was thinking about. It also speaks about deciding to not sync to NC storage some files like DVD images or so. 

I can imagine this is a difficult task to achieve, maybe not really technically as you are already using some sync protocols and such, but maybe more in the regards of how to put this together in a good way that is easy to use and understandable for the end user.

**Comment by michaelstingl** [comment]
Upstream project is experimenting with this in the [2.5.0 alpha1](https://owncloud.org/news/owncloud-client-goes-virtual-sync-files-demand/). Please help testing and provide feedback.

Main thread is here: https://github.com/nextcloud/desktop/issues/184

**Comment by intika** [comment]
I really do like the nextcloud desktop app but at the same time i miss a lot a one direction sync feature... So here is how to achieve one direction sync any way :D 

Just use nextcloud with firejail like so
```
firejail --noprofile --read-only=/what/ever/folder/you/are/syncing/subfolder/ /usr/bin/nextcloud
```
As easy as that :) and will avoid the hassle to loose files with a wrong sync direction
